File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # This normally requires some sort of dummy
        # the optimal solution will not convert to an array and back again

        prev = None
        curr = head

        while curr:
            nxt = curr.next
            curr.next = prev
            prev = curr
            curr = nxt

        return prev
        

File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int i = 0;

        while (i < nums.size() - 1) {
            if (nums[i] > nums[i+1]) {
                int dummy = nums[i+1];
                nums[i+1] = nums[i];
                nums[i] = dummy;
                i = 0;
            } else {
                i++;
            }
        }
        
    }
};